api/src/components/coauthor/service.ts

```diff
@@ -158,9 +158,16 @@ export const resetCoAuthors = async (publicationId: string) => {
     const resetCoAuthors = await client.prisma.coAuthors.updateMany({
         where: {
             publicationId,
-            NOT: {
-                linkedUser: publication?.createdBy
-            }
+            OR: [
+                {
+                    NOT: {
+                        linkedUser: publication?.createdBy
+                    }
+                },
+                {
+                    linkedUser: null
```

Comment:
This makes the invitation code (which is embedded into "Confirm & Review Publication" link ) reset every time the corresponding user unlocks the publications. The co-author cannot confirm or approve anymore because his initial invitation code is not valid anymore.

Steps to reproduce:

1. log in as a corresponding author and create a new publication
2. add a co-author
3. fill out everything and hit 'Request Approval' button
4. unlock the publication for changes (at this stage the invitation code for co-author is being reset)
5. hit 'Request Approval' button again
6. open a new browser instance and navigate to mailhog
7. try to accept the invitation that has been sent to co-author

<img width="1672" alt="image" src="https://github.com/JiscSD/octopus/assets/48569671/3eb220b2-6219-4352-add5-95d93fb10db1">

Reply:
Ah, I can see now why it was that way. At the time I thought this block was supposed to be catching null values too but I didn't realise the impact! I've reversed it.